corehq/apps/app_manager/suite_xml/post_process/workflow.py

```diff
@@ -332,16 +332,9 @@ def _get_link_frame(self, link, form, module):
         else:
             frame_children = _get_datums_matched_to_source(target_frame_children, source_form_datums)
 
+        # I think this is a bug - it only executes when linking to a child of the current module
```

Comment:
could you write a test to validate this?

Reply:
I did verify it locally - you mean a test to exercise the (presumably) flawed behavior?  That seems reasonable, and would make a fix easier to do when we're ready.